denguefever_tw/dengue_linebot/views.py

```diff
@@ -316,3 +316,63 @@ def suggestion_list(request):
 def gov_report_list(request):
     context = {'gov_reports': GovReport.objects.all().order_by('-report_time')}
     return render(request, 'dengue_linebot/gov_report_list.html', context)
+
+
+@login_required
+def push_msg_form(request):
+    context = {'areas': MinArea.objects.all()}
+    return render(request, 'dengue_linebot/push_msg.html', context)
+
+
+@login_required
+def push_msg_result(request):
+    areas_id = request.POST.getlist('msg_to')
+    content = request.POST['content']
+    img = request.POST['img']
+    error_msgs = list()
+    push_logs = list()
+
+    if not areas_id:
+        error_msgs.append('You do not choose any area!')
+    if not content and not img:
+        error_msgs.append('You do not write any content or image!')
+
+    if not error_msgs:
+        for area_id in areas_id:
+            users = list(LineUser.objects.filter(location=MinArea.objects.get(area_id=area_id)))
+            if users:
+                push_logs.extend(_push_msg(users=users, text=content, img=img))
+
+    return render(request, 'dengue_linebot/push_msg_result.html', {
+        'error_msgs':error_msgs,
+        'push_logs':push_logs
+    })
+
+
+def _push_msg(users, text, img):
+    splited_users_lists = _split_list(users)
+    msgs = list()
+    push_logs = list()
+
+    if text:
+        msgs.append(TextSendMessage(text=text))
+    if img:
+        msgs.append(ImageSendMessage(original_content_url=img, preview_image_url=img))
+
+    for users in splited_users_lists:
+        try:
+            line_bot_api.multicast([user.user_id for user in users], msgs)
+            push_logs.extend(["Successfully pushed msg to {user}".format(user=user) for user in users])
+        except LineBotApiError as e:
+            push_logs.extend(e.error.details)
+    return push_logs
```

Comment:
Sure. You can use `yield` to do the same thing by `yield` the content you extended to `push_logs`.

I think it would be fine to send a generator to django template.
But, I'm not sure. You can try it out.